Mathlib/RingTheory/Polynomial/Basic.lean

```diff
@@ -914,63 +914,61 @@ protected theorem Polynomial.isNoetherianRing [inst : IsNoetherianRing R] : IsNo
-          have : Submodule.span R[X] ↑s = Ideal.span ↑s := by rfl
-          rw [this]
-          intro p hp
-          generalize hn : p.natDegree = k
-          induction' k using Nat.strong_induction_on with k ih generalizing p
-          cases' le_or_lt k N with h h
-          · subst k
-            refine' hs2 ⟨Polynomial.mem_degreeLE.2
-              (le_trans Polynomial.degree_le_natDegree <| WithBot.coe_le_coe.2 h), hp⟩
-          · have hp0 : p ≠ 0 := by
-              rintro rfl
-              cases hn
-              exact Nat.not_lt_zero _ h
-            have : (0 : R) ≠ 1 := by
-              intro h
-              apply hp0
-              ext i
-              refine' (mul_one _).symm.trans _
-              rw [← h, MulZeroClass.mul_zero]
-              rfl
-            haveI : Nontrivial R := ⟨⟨0, 1, this⟩⟩
-            have : p.leadingCoeff ∈ I.leadingCoeffNth N := by
-              rw [HN]
-              exact
-                hm2 k
-                  ((I.mem_leadingCoeffNth _ _).2
-                    ⟨_, hp, hn ▸ Polynomial.degree_le_natDegree, rfl⟩)
-            rw [I.mem_leadingCoeffNth] at this
-            rcases this with ⟨q, hq, hdq, hlqp⟩
-            have hq0 : q ≠ 0 := by
-              intro H
-              rw [← Polynomial.leadingCoeff_eq_zero] at H
-              rw [hlqp, Polynomial.leadingCoeff_eq_zero] at H
-              exact hp0 H
-            have h1 : p.degree = (q * Polynomial.X ^ (k - q.natDegree)).degree := by
-              rw [Polynomial.degree_mul', Polynomial.degree_X_pow]
-              rw [Polynomial.degree_eq_natDegree hp0, Polynomial.degree_eq_natDegree hq0]
-              rw [Nat.cast_withBot, Nat.cast_withBot, Nat.cast_withBot, ← WithBot.coe_add,
-                add_tsub_cancel_of_le, hn]
-              · refine' le_trans (Polynomial.natDegree_le_of_degree_le hdq) (le_of_lt h)
-              rw [Polynomial.leadingCoeff_X_pow, mul_one]
-              exact mt Polynomial.leadingCoeff_eq_zero.1 hq0
-            have h2 : p.leadingCoeff = (q * Polynomial.X ^ (k - q.natDegree)).leadingCoeff := by
-              rw [← hlqp, Polynomial.leadingCoeff_mul_X_pow]
-            have := Polynomial.degree_sub_lt h1 hp0 h2
-            rw [Polynomial.degree_eq_natDegree hp0] at this
-            rw [← sub_add_cancel p (q * Polynomial.X ^ (k - q.natDegree))]
-            refine' (Ideal.span ↑s).add_mem _ ((Ideal.span ↑s).mul_mem_right _ _)
-            · by_cases hpq : p - q * Polynomial.X ^ (k - q.natDegree) = 0
-              · rw [hpq]
-                exact Ideal.zero_mem _
-              refine' ih _ _ (I.sub_mem hp (I.mul_mem_right _ hq)) rfl
-              rwa [Polynomial.degree_eq_natDegree hpq, Nat.cast_withBot, Nat.cast_withBot,
-                WithBot.coe_lt_coe, hn] at this
-            exact hs2 ⟨Polynomial.mem_degreeLE.2 hdq, hq⟩⟩⟩
+          have : x ∈ I.degreeLE N := hs ▸ Submodule.subset_span hx
+          this.2) <| by
+            have : Submodule.span R[X] ↑s = Ideal.span ↑s := by rfl
```

Comment:
this line and the rest of the block should be 4 spaces less

Reply:
Ah, I thought at first when reading "2 space indent" that it should be 2 spaces more, not less. Thanks.